pkg/controllers/localbuild/installer.go

```diff
@@ -0,0 +1,182 @@
+package localbuild
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+	"github.com/cnoe-io/idpbuilder/pkg/k8s"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var timeout = time.After(5 * time.Minute)
+
+type EmbeddedInstallation struct {
+	name         string
+	resourcePath string
+	namespace    string
+
+	// skips waiting on expected resources to become ready
+	skipReadinessCheck bool
+
+	// name and gvk pair for resources that need to be monitored
+	expectedResources map[string]schema.GroupVersionKind
```

Comment:
I think the name of this field should be `monitoredResources`. I first thought this was for making sure resources specified are present during installation.